### ISSUE TITLE
Raw Robj does not inherit the "raw" class

### ISSUE THREAD

**Issue by JosiahParry** [extendr/extendr#687]
Issue noticed in [extendr/b64#2](https://github.com/extendr/b64/pull/2).  `Robj`s have an `inherits()` method which is useful for checking classes of objects. However, the `Raw` object does not inherit the "raw" class as it should. 

```r
  rextendr::rust_function(r"-{
fn check_raw(x: Raw) -> bool {
  x.into_robj().inherits("raw")
}
}-")
#> ℹ build directory: '/private/var/folders/wd/xq999jjj3bx2w8cpg7lkfxlm0000gn/T/RtmpGPIl6H/filefb19695b5c7e'
#> ✔ Writing '/private/var/folders/wd/xq999jjj3bx2w8cpg7lkfxlm0000gn/T/RtmpGPIl6H/filefb19695b5c7e/target/extendr_wrappers.R'
  
  x <- raw()
  check_raw(x)
#> [1] FALSE
  inherits(x, "raw")
#> [1] TRUE
```

**Comment by yutannihilation** [comment]
"raw" is an implicit class. extendr's `inherits()` simply checks the `class` attribute, while R's `inherits()` considers implicit class. `?class` says:

> If the object does not have a class attribute, it has an implicit class, notably "matrix", "array", "function" or "numeric" or the result of typeof(x)

This is the C version of R's `ihnerits()`, but this isn't exposed as an API. So, there's no easy way to mimic the R's `inherits()`.

https://github.com/wch/r-source/blob/ec91103ee93b1d4e9f46a37c2dece3761192d9aa/src/main/objects.c#L1013

If the behavior is confusing, extendr might be better to consider removing it. But, I guess what you really want is `is_raw()` in this case, isn't it?

**Comment by JosiahParry** [comment]
Good to know! Thank you for sharing that. I suppose `is_raw()` is what I could be using here. I didn't know about it. In the use case I opted for `Raw::try_from()` and matched on the results which is sufficient. 

I think the `inherits()` is still useful so I don't think removal should be necessary. I've made a PR #689 to clarify the behavior of `inherits()` in the docstring.

**Comment by yutannihilation** [comment]
Can this be closed?

**Comment by CGMossa** [comment]
Can we please hold off on this? I want to propose a PR based on this.

**Comment by yutannihilation** [comment]
No problem (anyway I don't have permission to close, you know). I was just wondering if this was forgotten to be closed when #689 got merged.
